types/styled-system/index.d.ts

```diff
@@ -13,50 +13,63 @@
 //                 Chris LoPresto <https://github.com/chrislopresto>
 //                 Pedro Duarte <https://github.com/peduarte>
 //                 Dhalton Huber <https://github.com/Dhalton>
+//                 Elliot Bonneville <https://github.com/elliotbonneville>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 // TypeScript Version: 2.8
 
 import * as CSS from "csstype";
 
-export const defaultBreakpoints: string[];
-
-export function cloneFunc(fn: (...args: any[]) => any): (...args: any[]) => any;
 export function get(obj: any, ...paths: Array<string | number>): any;
-export function themeGet(keys: string, fallback?: string): any;
 
-export function is(n: any): boolean;
-export function isObject(n: any): boolean;
-export function num(n: any): boolean;
-export function px(n: any): string;
+export type Scale = object | Array<string | number>;
 
-export function createMediaQuery(n: number | string): string;
+// Preserved to support v4 shim:
+// https://github.com/styled-system/styled-system/blob/master/packages/styled-system/src/index.js#L108
+export interface LowLevelStyleFunctionArguments<N, S> {
+    prop: string;
+    cssProperty?: string;
+    alias?: string;
+    key?: string;
+    transformValue?: (n: N, scale?: S) => any;
+    scale?: S;
+    // new v5 api
+    properties?: string[];
+}
+
+export function style<N = string | number, S = Scale>(
+    // tslint:disable-next-line no-unnecessary-generics
+    args: LowLevelStyleFunctionArguments<N, S>
+): {
+    [cssProp: string]: string;
+};
 
 export interface styleFn {
     (...args: any[]): any;
-    propTypes?: string[];
 }
 
-export type Scale = object | Array<string | number>;
-
-export interface LowLevelStylefunctionArguments<N, S> {
-  prop: string;
-  cssProperty?: string;
-  alias?: string;
-  key?: string;
-  transformValue?: (n: N, scale?: S) => any;
-  scale?: S;
+export interface CustomStyleDefinition {
+    /** The CSS property to use in the returned style object (overridden by `properties` if present). */
+    property?: string;
```

Comment:
You can force here to use a valid CSS property, e.g.
```ts
property?: (keyof CSS.Properties)
```

Reply:
Awesome, I was looking for a way to do this!

Reply:
Done